### PR TITLE
EnvirGui: fix accidental overwrite of flag, SkipJack: fix stop method.

### PR DESCRIPTION
**Pull request by adcxyz**
## Purpose and Motivation

This fixes two related bugs: 
- EnvirGui:new always created a skipjack (accidental overwrite of the makeSkip flag)
- When a SkipJack stopped by its stopTest, it would not remove itself properly. 
 
- [x] Code is tested
- [x] All tests are passing
- [x] This PR is ready for review

```
///// Tests:

z = SkipJack({ "..".postln }, 1, { a == 0 }, \test);

// test stop with .stop method:
z.stop;
SkipJack.all.includes(z).not;  // true is OK
CmdPeriod.objects.includes(z).not; // true is OK


z.start;
// test stopping with stoptest
a = 0; // stoptest becomes true
SkipJack.all.includes(z).not;  // true is OK
CmdPeriod.objects.includes(z).not; // true is OK
```

### CODE REVIEW

File: SCClassLibrary/Common/Control/SkipJack.sc [hunk 1]
```diff
@@ -45,9 +45,9 @@ SkipJack {
 	play { this.start }
 
 	stop {
-		task.stop;
 		all.remove(this);
 		CmdPeriod.remove(this);
 		if( verbose ) { ("SkipJack" + name + "stopped.").postcln };
+		task.stop;
```
Comment (telephon) [inline SCClassLibrary/Common/Control/SkipJack.sc]: if we want to be exact, it's probably better to swap lines 50 and 51. If `task.stop` fails, the post will not lie.

Comment (adcxyz) [reply SCClassLibrary/Common/Control/SkipJack.sc]: ahm, no ;-) the problem is: 
the task stops itself from inside when stopTest becomes true; 
calling task.stop inside a task stops it right there, so the post would only happen if stopped from outside, not from inside. 
we could change it to ("SkipJack" + name + "stopping.") if you prefer.

Comment (telephon) [reply SCClassLibrary/Common/Control/SkipJack.sc]: yes, please! Then your explanation becomes much less necessary ;)

Comment (adcxyz) [reply SCClassLibrary/Common/Control/SkipJack.sc]: ok done!

**Comment by miguel-negrao** [comment]
@telephon Are you okay with merging this ?

**Comment by adcxyz** [comment]
I changed the post from "stopped" to "stopping", so IMO this is resolved, and ready to merge

**Comment by brianlheim** [comment]
@telephon We are in a code freeze. You know this. Do not merge any more pull requests.

**Comment by telephon** [comment]
sorry, I had assumed this concerned the c++ sources only.

**Comment by brianlheim** [comment]
It was announced twice and very clearly both times.

https://www.listarc.bham.ac.uk/lists/sc-dev/msg58864.html

https://github.com/supercollider/supercollider/pull/4348

This is a delicate operation and keeping both branches totally frozen makes it much easier to proceed confidently.

**Pull request merged**